### PR TITLE
Code 90 eslint best practices

### PR DESCRIPTION
**Pull request by NagaPradeepGajula-osi**
Configured the custom rules, turn no-console to warning, add babel plugin transform-remove-console and fixes

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -18,6 +18,35 @@ module.exports = {
   },
   rules: {
     'babel/no-invalid-this': 1,
+    'consistent-return': [2],
+    'dot-notation': [2],
+    'func-names': [2],
+    'jsx-a11y/label-has-for': [
```
Comment (eckdanny-osi) [inline .eslintrc.js]: just learned this rule is deprecated (its still present in `plugin:jsx-a11y/recommended` tho 😒). As long as its out of the way and either (`some`) id association or dom nesting is allowed, I'm cool

File: .eslintrc.js [hunk 2]
```diff
@@ -18,6 +18,35 @@ module.exports = {
...
+      'error',
+      {
+        components: ['label'],
+        required: {
+          some: ['nesting', 'id'],
+        },
+        allowChildren: true,
+      },
+    ],
+    'no-console': [1],
```
Comment (eckdanny-osi) [inline .eslintrc.js]: I'll open an issue to rake these from lib dists with `babel`. warning is perfect (annoy devs till its fixed, but don't break the build 🤣)

Comment (NagaPradeepGajula-osi) [reply .eslintrc.js]: I have the fixes for them locally will get them in a different PR. 
Stepped out to pick some tools to fix new home.

File: .eslintrc.js [hunk 3]
```diff
@@ -18,6 +18,35 @@ module.exports = {
...
+    'no-implicit-coercion': [2],
```
Comment (eckdanny-osi) [inline .eslintrc.js]: Lets keep the `!!` idiom. Way terser than `Boolean` constructor

```suggestion
    'no-implicit-coercion': [2, { "allow": ["!!"] } ],
```

File: .eslintrc.js [hunk 4]
```diff
@@ -18,6 +18,25 @@ module.exports = {
...
+    'no-magic-numbers': [1, {
+      ignore: [-1, 0, 1, 2]
+    }],
+    'no-useless-concat': [2],
+    'no-var': [2],
+    'prefer-const': [2],
+    'prefer-template': [2],
+    'radix': [2, 'as-needed'],
+    'react/display-name': [0],
```
Comment (eckdanny-osi) [inline .eslintrc.js]: landed in [#317](https://github.com/ca-cwds/design-system/pull/317/files#diff-e4403a877d80de653400d88d85e4801aR30). Dupe key?

File: package.json [hunk 5]
```diff
@@ -51,6 +51,7 @@
     "babel-plugin-transform-class-properties": "^6.24.1",
     "babel-plugin-transform-object-rest-spread": "^6.26.0",
     "babel-plugin-transform-rename-import": "^2.3.0",
+    "babel-plugin-transform-remove-console": "6.9.4",
```
Comment (eckdanny-osi) [inline package.json]: remove.

> I'll open an issue to rake these from lib dists with `babel`. warning is perfect (annoy devs till its fixed, but don't break the build 🤣)

Let that future PR own this (and the `rollup` configs to append the transform and whatever else is needed)

File: .eslintrc.js (hunk 1 above)
Comment (eckdanny-osi) [reply .eslintrc.js]: the deprecated rule is replaced by `label-has-associated-control` [link](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/59240febade673bdef3c5b2d1fdb289cb666a1bd/docs/rules/label-has-for.md#deprecated-label-has-for)

File: package.json (hunk 5 above)
Comment (NagaPradeepGajula-osi) [reply package.json]: 👍

Comment (NagaPradeepGajula-osi) [reply package.json]: 👍

File: .eslintrc.js (hunk 3 above)
Comment (NagaPradeepGajula-osi) [reply .eslintrc.js]: 👍

Comment (NagaPradeepGajula-osi) [reply .eslintrc.js]: done

File: .eslintrc.js (hunk 4 above)
Comment (NagaPradeepGajula-osi) [reply .eslintrc.js]: probably

File: package.json (hunk 5 above)
Comment (NagaPradeepGajula-osi) [reply package.json]: done

File: .eslintrc.js (hunk 1 above)
Comment (NagaPradeepGajula-osi) [reply .eslintrc.js]: done

File: .eslintrc.js (hunk 2 above)
Comment (NagaPradeepGajula-osi) [reply .eslintrc.js]: Done in PR #324

File: apps/www/src/modules/labs/views/DataGridsPage/AdvancedSearch/mockService.js [hunk 6]
```diff
@@ -1,17 +1,17 @@
 exports.__esModule = true
```
Comment (eckdanny-osi) [inline apps/www/src/modules/labs/views/DataGridsPage/AdvancedSearch/mockService.js]: this must be a typescript compilation artifact. No action required... but we should clean up the TS stuff soon if we care enough to lint the output.

File: libs/components/src/Rolodex/Rolodex.test.js [hunk 7]
```diff
@@ -24,14 +33,16 @@ describe('Rolodex', () => {
...
       expect(() =>
-        mount(
-          <Rolodex>
-            <Card>
-              <CardHeader>Good</CardHeader>
-              <CardBody>Good</CardBody>
-            </Card>
-            <div>Bad</div>
-          </Rolodex>
+        mount( <
```
Comment (eckdanny-osi) [inline libs/components/src/Rolodex/Rolodex.test.js]: um.... not sure what happened here.

Comment (NagaPradeepGajula-osi) [reply libs/components/src/Rolodex/Rolodex.test.js]: seems like auto save white spacing. fix is on the way

Comment (NagaPradeepGajula-osi) [reply libs/components/src/Rolodex/Rolodex.test.js]: done

**Comment by eckdanny-osi** [comment]
changed base branch for all outstanding eslint-related PRs (#320, #323, #324, #321)

**Pull request merged**